lib/Slack_web_api.js

```diff
@@ -1,314 +1,196 @@
-            },
-            get: function(options, cb) {
-                slack_api.callAPI('reactions.get', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('reactions.list', options, cb);
-            },
-            remove: function(options, cb) {
-                slack_api.callAPI('reactions.remove', options, cb);
-            },
-        },
-        rtm: {
-            start: function(options, cb) {
-                slack_api.callAPI('rtm.start', options, cb);
-            },
-        },
-        search: {
-            all: function(options, cb) {
-                slack_api.callAPI('search.all', options, cb);
-            },
-            files: function(options, cb) {
-                slack_api.callAPI('search.files', options, cb);
-            },
-            messages: function(options, cb) {
-                slack_api.callAPI('search.messages', options, cb);
-            },
-        },
-        stars: {
-            list: function(options, cb) {
-                slack_api.callAPI('stars.list', options, cb);
-            },
-        },
-        team: {
-            accessLogs: function(options, cb) {
-                slack_api.callAPI('team.accessLogs', options, cb);
-            },
-            info: function(options, cb) {
-                slack_api.callAPI('team.info', options, cb);
-            },
-        },
-        users: {
-            getPresence: function(options, cb) {
-                slack_api.callAPI('users.getPresence', options, cb);
-            },
-            info: function(options, cb) {
-                slack_api.callAPI('users.info', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('users.list', options, cb);
-            },
-            setActive: function(options, cb) {
-                slack_api.callAPI('users.setActive', options, cb);
-            },
-            setPresence: function(options, cb) {
-                slack_api.callAPI('users.setPresence', options, cb);
-            },
         }
+        slack_api.callAPI('chat.postMessage', options, cb);
     };
 
     return slack_api;
 
+
+    /**
+     * Makes a POST request as a form to the given url with the options as data
+     * @param {string} url The URL to POST to
+     * @param {Object} formData The data to POST as a form
+     * @param {function=} cb An optional NodeJS style callback when the POST completes or errors out.
+     */
+    function postForm(url, formData, cb) {
+        cb = cb || noop;
+
+        bot.log('** API CALL: ' + url);
+        request.post(url, function(error, response, body) {
+            bot.debug('Got response', error, body);
+            if (!error && response.statusCode == 200) {
+                var json;
+                try {
+                    json = JSON.parse(body);
+                } catch (parseError) {
+                    return cb(parseError);
```

Comment:
The first line checks for a callback and if it's not defined sets it to the noop function. That simplifies the checks later since cb will always be defined.